### PR TITLE
RavenDB-17492 : stop replication on attempt to send incremental time series to an old server

### PR DESCRIPTION
**Pull request by aviv86**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17492

### Additional description

- add `IncrementalTimeSeries` to `SupportedFeatures.Replication`
- stop replication and throw an error on attempt to replicate incremental time series to an old node with a lower protocol version
- add inter version test

### CODE REVIEW

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs [hunk 1]
```diff
@@ -425,6 +428,11 @@ private bool CanContinueBatch(ReplicationState state, ref long next)
                 AssertNotClusterTransactionDocumentForLegacyReplication(state.Item);
             }
 
+            if (_parent.SupportedFeatures.Replication.IncrementalTimeSeries == false)
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs]: We enter here only if `replicationState.LastTransactionMarker != item.TransactionMarker`, when I think about it all of those assertion are in the wrong place?

What if I change a doc and increment a TS in the same tx, I will not fail here, right?

Comment (aviv86) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs]: you are correct
added a failing test : https://github.com/ravendb/ravendb/pull/13040/commits/db788a7a35d85ea055dd7ff0dcb942c1ddecf0d7

Comment (karmeli87) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs]: We need to fix this and backport the fix... all of those assertions will not be triggered under the above conditions

Comment (aviv86) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs]: https://github.com/ravendb/ravendb/pull/13056

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs [hunk 2]
```diff
@@ -507,6 +520,42 @@ private void AssertNotClusterTransactionDocumentForLegacyReplication(Replication
             }
         }
 
+        private void AssertNotIncrementalTimeSeriesForLegacyReplication(ReplicationBatchItem item)
+        {
+            if (item.Type == ReplicationBatchItem.ReplicationItemType.TimeSeriesSegment || item.Type == ReplicationBatchItem.ReplicationItemType.DeletedTimeSeriesRange)
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:525]: Reversing the `if` will reduce nesting

**Review by ayende — APPROVED** [review]

**Pull request merged**